ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -605,4 +605,143 @@ - (NSArray*)serializePacksArray:(NSArray<MGLOfflinePack*>*)packs
+        coordinates = (NSMutableArray *) @[@(feature.coordinate.longitude), @(feature.coordinate.latitude)];
+    } else if ([feature isKindOfClass:[MGLPolylineFeature class]]) {
+        geometryType = @"LineString";
+        MGLPolylineFeature *polyline = (MGLPolylineFeature *)feature;
+        coordinates = [self getMGLPolylineCoordinates:polyline];
+    } else if ([feature isKindOfClass:[MGLPolygonFeature class]]) {
+        geometryType = @"Polygon";
+        MGLPolygonFeature *polygon = (MGLPolygonFeature *)feature;
+        coordinates = [self getMGLPolygonCoordinates:polygon];
+    } else if ([feature isKindOfClass:[MGLMultiPolylineFeature class]]) {
+        geometryType = @"MultiLineString";
+        MGLMultiPolylineFeature *multiPolyline = (MGLMultiPolylineFeature *)feature;
+        for (MGLPolylineFeature *polyline in multiPolyline.polylines) {
+            [coordinates addObject:[self getMGLPolylineCoordinates:polyline]];
+        }
+    } else if ([feature isKindOfClass:[MGLMultiPolygonFeature class]]) {
+        geometryType = @"MultiPolygon";
+        MGLMultiPolygonFeature *multiPolygon = (MGLMultiPolygonFeature *)feature;
+        for (MGLPolygonFeature *polygon in multiPolygon.polygons) {
+            [coordinates addObject:[self getMGLPolygonCoordinates:polygon]];
+        }
+    } else if ([feature isKindOfClass:[MGLMultiPointFeature class]]) {
+        // this is checked last since MGLPolyline and MGLPolygon inherit from MGLMultiPoint
+        geometryType = @"MultiPoint";
+        MGLMultiPointFeature *multiPoint = (MGLMultiPointFeature *)feature;
+        for (int index = 0; index < multiPoint.pointCount; index++) {
+            CLLocationCoordinate2D coord = multiPoint.coordinates[index];
+            [coordinates addObject:@[@(coord.longitude), @(coord.latitude)]];
+        }
+    }
+
+    return @{ @"type": geometryType,
+              @"coordinates": coordinates };
+}
+
+- (NSMutableArray *)getMGLPolylineCoordinates:(MGLPolylineFeature *)polyline
+{
+    NSMutableArray *coordinates = [[NSMutableArray alloc] init];
+    for (int index = 0; index < polyline.pointCount; index++) {
+        CLLocationCoordinate2D coord = polyline.coordinates[index];
+        [coordinates addObject:@[@(coord.longitude), @(coord.latitude)]];
+    }
+    return coordinates;
+}
+
+- (NSMutableArray *)getMGLPolygonCoordinates:(MGLPolygonFeature *)polygon
+{
+    NSMutableArray *coordinates = [[NSMutableArray alloc] init];
+
+    NSMutableArray *outerRingCoordinates = [[NSMutableArray alloc] init];
+    for (int index = 0; index < polygon.pointCount; index++) {
+        CLLocationCoordinate2D coord = polygon.coordinates[index];
+        [outerRingCoordinates addObject:@[@(coord.longitude), @(coord.latitude)]];
+    }
+    [coordinates addObject:outerRingCoordinates];
+
+    for (MGLPolygonFeature *interiorRing in polygon.interiorPolygons) {
+        NSMutableArray *interiorRingCoordinates = [[NSMutableArray alloc] init];
```

Comment:
Wouldn't calling it recursively nest the arrays too much?
```
[outerRing, interiorRing1, interiorRing2, ...]
```
vs
```
[outerRing, [interiorRing1], [interiorRing2], ...]
```

I've left it as is in my refactoring to using a category.